blenderproc/python/loader/ObjectLoader.py

```diff
@@ -23,47 +28,79 @@ def load_obj(filepath: str, cached_objects: Optional[Dict[str, List[MeshObject]]
     :param kwargs: all other params are handed directly to the bpy loading fct. check the corresponding documentation
     :return: The list of loaded mesh objects.
     """
-    if os.path.exists(filepath):
-        if cached_objects is not None and isinstance(cached_objects, dict):
-            if filepath in cached_objects.keys():
-                created_obj = []
-                for obj in cached_objects[filepath]:
-                    # duplicate the object
-                    created_obj.append(obj.duplicate())
-                return created_obj
-            loaded_objects = load_obj(filepath, cached_objects=None, **kwargs)
-            cached_objects[filepath] = loaded_objects
-            return loaded_objects
-        # save all selected objects
-        previously_selected_objects = bpy.context.selected_objects
-        if filepath.endswith('.obj'):
-            # load an .obj file:
-            if use_legacy_obj_import:
-                bpy.ops.import_scene.obj(filepath=filepath, **kwargs)
-            else:
-                bpy.ops.wm.obj_import(filepath=filepath, **kwargs)
-        elif filepath.endswith('.ply'):
+    if not os.path.exists(filepath):
+        raise FileNotFoundError(f"The given filepath does not exist: {filepath}")
+
+    if cached_objects is not None and isinstance(cached_objects, dict):
+        if filepath in cached_objects.keys():
+            created_obj = []
+            for obj in cached_objects[filepath]:
+                # duplicate the object
+                created_obj.append(obj.duplicate())
+            return created_obj
+        loaded_objects = load_obj(filepath, cached_objects=None, **kwargs)
+        cached_objects[filepath] = loaded_objects
+        return loaded_objects
+    # save all selected objects
+    previously_selected_objects = bpy.context.selected_objects
+    if filepath.endswith(".obj"):
+        # load an .obj file:
+        if use_legacy_obj_import:
+            bpy.ops.import_scene.obj(filepath=filepath, **kwargs)
+        else:
+            bpy.ops.wm.obj_import(filepath=filepath, **kwargs)
+    elif filepath.endswith(".ply"):
+        PLY_TEXTURE_FILE_COMMENT = "comment TextureFile "
+        model_name = os.path.basename(filepath)
+
+        # Read file
+        with open(filepath, "r", encoding="latin-1") as file:
```

Comment:
Why is the encoding set here? Are ply files always latin-1 encoded?

Reply:
Previous encofing was utf-8, but this lead to some errors with some of the .ply files. Changing it to latin-1 solved the issue and allowed to load all .ply models without issue. Not sure though why some of the models couldn't be imported with utf-8, the once I created myself in blender couldn't be loaded that way.